nsv13/code/game/objects/items/bloody_eye.dm

```diff
@@ -0,0 +1,192 @@
+/obj/item/reagent_containers/hypospray/bloody_eye
+	name = "\improper Bloody Eye spray"
+	desc = "This vial of Bloody Eye is equipped with a spray nozzle and a handy measuring ring to make sure the spray cone is just the right size."
+	icon = 'nsv13/icons/obj/nsv13_syringe.dmi'
+	item_state = "bloody"
+	lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
+	icon_state = "bloody"
+	amount_per_transfer_from_this = 30
+	volume = 30
+	possible_transfer_amounts = list(30)
+	resistance_flags = ACID_PROOF
+	reagent_flags = OPENCONTAINER
+	slot_flags = ITEM_SLOT_BELT
+	list_reagents = list(/datum/reagent/drug/bloody_eye = 30)
+
+/datum/reagent/drug/bloody_eye
+	name = "Bloody Eye"
+	description = "A powerful drug of Syndicate origin. Manufactured to give the user an incredibly exciting rush, typically exilerating the user into a state of rage-induced psychosis. Boosts physical strength and allows the user to ignore the effects of critical condition."
+	reagent_state = LIQUID
+	color = "#D92323"
+	overdose_threshold = 5
+	addiction_threshold = 5
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+
+/datum/reagent/drug/bloody_eye/on_mob_metabolize(mob/living/L)
+	ADD_TRAIT(L,TRAIT_MONKEYLIKE,type)
+	ADD_TRAIT(L,TRAIT_NOCRITDAMAGE,type)
+	ADD_TRAIT(L,TRAIT_NOSOFTCRIT,type)
+	ADD_TRAIT(L,TRAIT_NOHARDCRIT,type)
+	ADD_TRAIT(L,TRAIT_NOBLOCK,type)
+	ADD_TRAIT(L,TRAIT_STUNIMMUNE,type)
+	ADD_TRAIT(L,TRAIT_SLEEPIMMUNE,type)
+	ADD_TRAIT(L,TRAIT_IGNOREDAMAGESLOWDOWN,type)
+	ADD_TRAIT(L,TRAIT_NOSTAMCRIT,type)
+	ADD_TRAIT(L,TRAIT_NOLIMBDISABLE,type)
```

Comment:
You can put all of the trait handling in the status effect too

Reply:
Done